tests/ui/type-alias-impl-trait/different_defining_uses_never_type-2.rs

```diff
@@ -0,0 +1,15 @@
+#![feature(type_alias_impl_trait)]
```

Comment:
also add test with type params for https://github.com/rust-lang/rust/issues/110601 and close that issue, i think this should fix it?

Reply:
We actually don't need to include this commit anymore. It's actually fine that our HIR typeck results have a `Map<DefId, HiddenTy>`, since we do the mapping from hidden ty to definition params during writeback:

https://github.com/rust-lang/rust/blob/8b4b20836b832e91aa605a2faf5e2a55190202c8/compiler/rustc_hir_typeck/src/writeback.rs#L582-L588

So I'll just land #111853 independently of this PR.